capable/src/lib.rs

```diff
@@ -1,3 +1,85 @@
 // Copyright (c) 2022 MobileCoin Foundation
 
 #![doc = include_str!("../README.md")]
+
+use mc_sgx_capable_sys_types::sgx_device_status_t;
+use mc_sgx_capable_types::{Error, Result};
+use mc_sgx_core_sys_types::sgx_status_t;
+use mc_sgx_core_types::Error as SgxError;
+
+/// Tests if the current system is capable of running SGX.
+///
+/// This function requires running the untrusted userspace application as an
+/// administrator.
+pub fn is_capable() -> Result<bool> {
+    let mut value = 0;
+    let status = unsafe { mc_sgx_capable_sys::sgx_is_capable(&mut value as *mut _) };
+    if status == sgx_status_t::SGX_SUCCESS {
+        Ok(value == 1)
+    } else {
+        Err(Error::from(SgxError::from(status)))
```

Comment:
What's the value of using `mc_sgx_capable_types::Error` here when you're always going to get the `Sgx(SgxError)` variant? Why not just make `SgxError` the error type for this function's `Result`?

Alternately, if `Sgx(SgxError)` is only for the `SGX_ERROR_NO_PRIVILEGE` case, why not name it something more descriptive, e.g. `UefiPermissions`, or `UefiPermissions(SgxError)` if you want to preserve the `SgxError` just in case.

Reply:
The basic intent is to make all the methods in `mc_sgx_capable` return `mc_sgx_capable::Error`. This function could return `SgxError`, but it would be the odd one out for the rest of them...